### DISCUSSION TITLE
Intellisence is very slow when using VS Code

### DISCUSSION THREAD

**Discussion by m-haketa** [remarkjs/remark#1023] [Q&A] [answered]
Intellisence is very slow when using VS Code. Is there anything I can do to make it faster?

It takes about 2 seconds to get the result of intellisence. It becomes worse when I install and import other plugins ( for example, remark-directive) 

### package.json
```json
{
  "name": "test",
  "dependencies": {
    "unified": "^10.1.2",
    "unist-util-visit": "^4.1.0"
  },
  "devDependencies": {
    "@types/mdast": "^3.0.3",
    "typescript": "^4.7.4"
  },
  "type": "module"
}
```

### tsconfig.json
```json
{
  "compilerOptions": {
    "module": "esnext",
    "target": "esnext",
    "lib": ["esnext", "DOM"],
    "strict": true,
    "esModuleInterop": true,
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "baseUrl": ".",
    "sourceMap": true
  }
}
```

### src/index.ts
```ts
import { visit } from 'unist-util-visit';

const myRemarkPlugin: import('unified').Plugin<[], import('mdast').Root> = () => (tree) => {
  visit(tree, (node) => {
    //node.   <- very slow
  });
};
```

**Comment by wooorm** [comment, marked as answer]
Right, `visit` can be slow.
You can help it by passing a test.
And, split things up in different files, so that the file using expensive operations doesn‘t affect others.

PRs to `unist-util-visit`’s fancy types to improve this are welcome! Practically, it would be these types: https://github.com/syntax-tree/unist-util-visit-parents/blob/c4fdbdbdf9ce1a4549d2d6ba32a1f0eef92cf237/complex-types.d.ts#L43-L86.
